Surrealdb V3 support

rynoV opened the pull request.

rynoV committed:
surrealdb 3.0.0-alpha

rynoV committed:
justfile update

rpfontana commented:
Hi @rynoV, I'm using this crate in a project and I'd like to move to surrealdb v3.  

Are you still working on this? I can help out if needed, thanks anyway

rynoV commented:
Oh yeah feel free to open a PR into this branch. IIRC this was working with the alpha, hopefully there aren't too many differences from there

nresare committed:
Test fixes: kv-mem feature is needed, define table on setup

With the arrival of SurrealDB 3, strict mode is now the default
and inserting into a table before it is defined fails.

dev-dependencies also needs a version of surrealdb that
includes the kv-mem feature.

nresare commented:
I have now successfully used this branch with one o my projects with some small tweaks and updates that I opened in separate PRs against this branch and it seems to work at least for the simple cases

nresare committed:
Allow sharing Surreal client with session store

rynoV committed:
Remove surrealdb-nightly

afaict it isn't supported anymore

rynoV committed:
Remove surrealdb feature

No longer needed now that nightly is gone

rynoV committed:
Update to surrealdb v3 stable

rynoV committed:
Formatting

rynoV committed:
Simplify build commands

rynoV committed:
Merge remote-tracking branch 'nresare/shared_surreal_instance' into surrealdb-v3

rynoV committed:
Minor docs updates

rynoV committed:
Use Arc::clone for clarity

rynoV committed:
Merge remote-tracking branch 'origin/master' into surrealdb-v3

rynoV committed:
Fixing minimum versions

rynoV committed:
Note about removed `surrealdb-nightly` feature

rynoV committed:
Remove minimal-versions checks from CI
